configure.ac

```diff
@@ -357,7 +357,9 @@ case $host in
 esac
 
 if test "$enable_debug" = "yes"; then
-  dnl Clear default -g -O2 flags
+  dnl If debugging is enabled, and the user hasn't overriden CXXFLAGS, clear
+  dnl them, to prevent autoconfs "-g -O2" being added. Otherwise we'd end up
+  dnl with "-O0 -g3 -g -O2".
```

Comment:
Probably we should differentiate between necessary flags (eg, include paths) and provided defaults (eg, -O2 -g), and do "$DEFAULTS $USER $NECESSARY" so this just works without needing a special hack.

Reply:
I think the philosophy (also according to [automake docs](https://www.gnu.org/software//automake/manual/html_node/Flag-Variables-Ordering.html)) is that the user has always the last word, so it should really be "$DEFAULTS $USER $NECESSARY". This gives the user also the last word on what is "necessary", e.g., the user may have a strange compiler/wrapper, where other flags are necessary. 

If the user wants to break the build by overriding flags, that's his choice. And even this can be helpful for debugging the build.

Reply:
> so it should really be "$DEFAULTS $USER $NECESSARY".

I assume you mean "$DEFAULTS $NECESSARY $USER" ?

I agree. The point of these change is to stop overriding user choice, so doing something that would prevent the user from having the final say would defeat the point.

Reply:
> I assume you mean "$DEFAULTS $NECESSARY $USER" ?

Sure, sorry, copy and paste...